l1-contracts/contracts/governance/L2AdminFactory.sol

```diff
@@ -21,12 +22,14 @@ contract L2AdminFactory {
     address[] public requiredRestrictions;
 
     constructor(address[] memory _requiredRestrictions) {
+        _validateRestrctions(_requiredRestrictions);
         requiredRestrictions = _requiredRestrictions;
     }
 
     /// @notice Deploys a new L2 admin contract.
     /// @return admin The address of the deployed admin contract.
     function deployAdmin(address[] calldata _additionalRestrictions, bytes32 _salt) external returns (address admin) {
```

Comment:
Just because `_validateRestrctions` already use memory
```suggestion
    function deployAdmin(address[] memory _additionalRestrictions, bytes32 _salt) external returns (address admin) {
```